### PR TITLE
4. Documentation change - section Tutorial : Using Interfaces

### PR DESCRIPTION
**Pull request by AroosaIjaz**
This PR contains only the third section of the tutorials which has one example. 

1. I am thinking of implementing another example across different interfaces. So there might be an addition here later on.

### CODE REVIEW

File: doc/tutorials/isingmodel.ipynb [hunk 1]
```diff
@@ -0,0 +1,612 @@
+{
```
Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>```We will be comparing PennyLane's <em>gradient descent optimizer</em> with that of <code>Tensorflow</code> and <code>Pytorch</code> for one and the same quantum system. ```</p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>One is red and one is gray?</p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>Let's test these functions for the [1,-1,-1] spin configuration. </p><p>The total energy for this Ising model should be </p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>0,np.pi -&gt; 0, np.pi</p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>"using the"</p><p><br></p><p>maybe mention before that the base one is the NumPy one?</p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>Maybe: "The minimun energy should be -2 for the [1,1,-1] spin configuration, which corresponds to angles of [0,0,0] and [0,pi,0] for the spin 2 and spin 3 respectively."</p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>Delete or surpress the warning?</p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>Maybe rearrange long lines of code to separate lines? </p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>Here and elsewhere: Python beauty standard has a space before comma</p><p><br></p><p>circuit3(var1, var2)</p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (mariaschuld) [inline doc/tutorials/isingmodel.ipynb]: <p>Mention, what is closure?</p>

 _Reply via <a href='https://app.reviewnb.com/XanaduAI/pennylane/pull/219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (AroosaIjaz) [reply doc/tutorials/isingmodel.ipynb]: <p>on it. </p>

**Comment by co9olguy** [comment]
@josh146 @AroosaIjaz now that new gallery extension is merged into `doc-refactor` branch, we'll have to revisit these PRs, especially to see if any further tweaks need to be done

**Pull request closed without merging**